knora/dsplib/utils/excel_to_json_lists.py

```diff
@@ -216,36 +248,34 @@ def make_root_node_from_args(excelfiles: list[str], listname_from_args: str, com
 
     Returns:
         dict: The root node of the list as dictionary (JSON)
-
     """
-    rootnode_labels_dict = {}
-    listname = listname_from_args
-    listname_en = ''
+    lang_code_2_listname = {}
+    listname_en: str = ''
+    lang_specific_listname: str = ''
 
     for filename in excelfiles:
         basename = os.path.basename(filename)
-        label, lang_code = os.path.splitext(basename)[0].rsplit('_', 1)
+        lang_specific_listname, lang_code = os.path.splitext(basename)[0].rsplit('_', 1)
 
-        # check if language code is valid
         if not check_language_code(lang_code):
-            print('Invalid language code is used. Only language codes from ISO 639-1 and ISO 639-2 are accepted.')
+            print(f'Invalid language code "{lang_code}" is used. Only language codes from ISO 639-1 ',
+                  f'and ISO 639-2 are accepted.')
             quit()
 
-        rootnode_labels_dict[lang_code] = label
-        listname = label
+        lang_code_2_listname[lang_code] = lang_specific_listname
 
         if '_en.xlsx' in filename:
-            listname_en = label
-
-    # if an english list is available use its label as listname
-    if listname_en:
-        listname = listname_en
+            listname_en = lang_specific_listname
 
-    # if the user provided a listname use it
-    if listname_from_args:
+    # the listname is taken from the following sources, with descending priority
+    if isinstance(listname_from_args, str) and listname_from_args != '':
```

Comment:
You don't have to check if `listname_from_args` is a string as this is a method parameter defined as string. If you use `if listname_from_args` (as it was before) it returns False in case of an empty string.

Reply:
And what happens if `isinstance(listname_from_args, str)` is not defined? It's type is `Optional[str]`. But you're right, my version is not so good...

Reply:
Optional means it is either a string or `None`. If it is `None`, `if listname_from_args` would return `False`.
Does that answer your question?

Reply:
yes, thanks. I changed it back.